lib/auth.js

```diff
@@ -84,6 +97,57 @@ var securityDefinitionCreators = {
                 });
             }
         };
+    },
+    header_match: function(definition) {
+        var errorMessage = definition['x-error-message']
+            || 'This client is not allowed to use the endpoint';
+        var whitelistMap = {};
+        Object.keys(definition['x-whitelists']).forEach(function(whitelistName) {
+            whitelistMap[whitelistName]
+                = constructInternalRequestRegex(definition['x-whitelists'][whitelistName]);
+        });
+
+        return {
+            prepareRequest: function() {
+            },
+            checkPermissions: function(restbase, req, permissions) {
+                if (restbase._rootReq.uri === '#internal-startup') {
+                    // Skip a check on requests made by RESTBase during startup
+                    return;
+                }
+
+                permissions.forEach(function(requirementDefinition) {
+                    // Check if requirement is limited to some method
+                    if (requirementDefinition.method
+                            && requirementDefinition.method !== req.method) {
+                        return;
+                    }
+
+                    var headerMatchRequirement = requirementDefinition.value;
+                    var headerName = headerMatchRequirement.header;
+                    var headerValue = req.headers && req.headers[headerName]
+                            || restbase._rootReq.headers && restbase._rootReq.headers[headerName];
+
+                    headerMatchRequirement.patterns.forEach(function(permission) {
```

Comment:
The use of 'permission' seems to be a bit confusing here. Do you think 'pattern' would be clearer?